### PR TITLE
Rub 26

### PR DESCRIPTION
**Pull request by intsrc**
# REVIEWERS

- [ ] @sonishc 
- [ ] @vlodkopits 

#

# DESCRIPTION

Added ability for user to use items in the game. Currently, only healing is available as a spell effect for items.

#

# EXPERT REVIEW

- [ ] @rokor
- [ ] @okohuch

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    current_user.increment(:experience, params[:experience])
+
+    if current_user.save
+      render json: { status: 'OK' }.to_json
+    else
+      render json: current_user.errors, status: :unprocessable_entity
+    end
+  end
+
+  def remove_item
```
Comment (okohuch) [inline app/controllers/users_controller.rb]: Why is this action in user controller? Should not it be in Items controller?

File: app/assets/javascripts/components/fight.jsx [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    this.setState({
+      botHit: {
+        selectedAttackIndex: attack,
+        selectedBlockIndex: block
+      }
+    });
+  }
+
+  initAttack(attack, block, player) {
+    if ( attack != block) {
```
Comment (rokor) [inline app/assets/javascripts/components/fight.jsx:66]: if inside if inside if try to find another solution

File: app/assets/javascripts/components/fight.jsx [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+  getDamage(actor) {
+    let max = 0, armor_bonus = 0;
+    if (actor == 'user') {
+      max = USER_DAMAGE;
+    } else {
+      armor_bonus = Math.floor(this.state.user.armor * ARMOR_MULTIPLIER);
+      max = BOT_DAMAGE;
+    }
+    const random = this.getRandomInt(1, max);
```
Comment (rokor) [inline app/assets/javascripts/components/fight.jsx:85]: let should be fine 85-86

File: app/assets/javascripts/components/fight.jsx [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+    const {
+      selectedAttackIndex: selectedBotAttackIndex,
+      selectedBlockIndex: selectedBotBlockIndex
+    } = this.state.botHit;
+
+    this.initAttack(selectedAttackIndex, selectedBotBlockIndex, 'Bot')
+    this.setAnimationLog(HITS_ANIMATIONS_URL.hit, LOGS.userLog)
+
+    const promise = new Promise(resolve => {
+        setTimeout(() => resolve(this.buttonEnabled(true)), 4000);
```
Comment (rokor) [inline app/assets/javascripts/components/fight.jsx:109]: setTimeout with Promise??

**Pull request closed without merging**